cmd/web-handlers.go

```diff
@@ -648,7 +650,39 @@ func (web *webAPIHandlers) Upload(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	hashReader, err := hash.NewReader(r.Body, size, "", "")
+	reader := r.Body
+	var hashError error
+	actualSize := size
+
+	if objectAPI.IsCompressionSupported() && isCompressible(r.Header, object) && size > 0 {
+		// Storing the compression metadata.
+		metadata[ReservedMetadataPrefix+"compression"] = compressionAlgorithmV1
+		metadata[ReservedMetadataPrefix+"actual-size"] = strconv.FormatInt(size, 10)
```

Comment:
I am using it back in `GetActualSize()` . So, i need this meta info for fs uploads. They dont have PartsInfo for small sized uploads. like here ,

![image](https://user-images.githubusercontent.com/5410427/45463089-65d78c80-b728-11e8-9bf0-341428e1a84a.png)

So GetActualSize function reads from the meta if the parts info is not present

Reply:
I was thinking that we should now start storing that information in the PartsInfo{} such that we can avoid storing multiple ways in the same file. PartInfo can be filled always on all objects which are stored with compression. 

This will allow us to avoid using multiple ways to handle this situation.

Reply:
ok should we change the `fs.putobject` to save partsInfo even for small size uploads ?

Reply:
Done